app/static/app/js/components/TaskList.jsx

```diff
@@ -1,23 +1,68 @@
 import React from 'react';
 import '../css/TaskList.scss';
+import TaskListItem from './TaskListItem';
 
 class TaskList extends React.Component {
   constructor(){
     super();
+
+    this.state = {
+      tasks: [],
+      error: "",
+      loading: true
+    };
+
+    this.refresh = this.refresh.bind(this);
   }
 
   componentDidMount(){
-    console.log("DISPLAY");
+    this.loadTaskList(); 
   }
 
   refresh(){
-    console.log("REFRESH");
+    this.loadTaskList();
+  }
+
+  loadTaskList(){
+    this.taskListRequest = 
+      $.getJSON(this.props.source, json => {
+          this.setState({
+              tasks: json
+          });
+        })
+        .fail((jqXHR, textStatus, errorThrown) => {
+          this.setState({ 
+              error: `Could not load projects list: ${textStatus}`,
+          });
+        })
+        .always(() => {
+          this.setState({
+            loading: false
+          })
+        });
+  }
+
+  componentWillUnmount(){
+    this.taskListRequest.abort();
   }
 
   render() {
+    let message = "";
+    if (this.state.loading){
+      message = (<span>Loading... <i className="fa fa-refresh fa-spin fa-fw"></i></span>);
+    }else if (this.state.error){
+      message = (<span>Could not get tasks: {this.state.error}. <a href="javascript:void(0);" onClick={this.refresh}>Try again</a></span>);
+    }else if (this.state.tasks.length === 0){
+      message = (<span>This project has no tasks. Create one by uploading some images!</span>);
+    }
```

Comment:
Ah, neat! I've never seen the switch used that way. If I add a forth else if I will switch to that syntax (usually 3 if-else-if combinations are bearable to read).